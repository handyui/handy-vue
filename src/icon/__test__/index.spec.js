import {mount, shallowMount} from '@vue/test-utils'
import Icon from '../src/index.vue'

describe('Icon.vue', ()=>{
  it('renders props.msg when passed', ()=>{
    const wrapper = mount(Icon, {
      propsData: {
        name: 'success',
      },
    });
    expect(wrapper).toMatchSnapshot();
  })

  it('renders props.msg when passed', ()=>{
    const msg = 'new icon'
    const wrapper = shallowMount(Icon, {
      propsData: { msg }
    })
    expect(wrapper.findAll('.ha-icon').length).toBe(1)
  })

  it('dot prop', () => {
    const wrapper = mount(Icon, {
      propsData: {
        dot: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
})