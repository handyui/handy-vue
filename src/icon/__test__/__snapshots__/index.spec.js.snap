// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue dot prop 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <i
        class="ha-icon ha-icon-default"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </i>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.5",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "badge": "",
    "classes": Array [
      "ha-icon",
      Object {
        "ha-icon-default": true,
      },
    ],
    "color": "",
    "dot": false,
    "imageIcon": false,
    "name": "default",
    "prefixCls": "ha-icon",
    "size": "default",
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;

exports[`Icon.vue renders props.msg when passed 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <i
        class="ha-icon ha-icon-default"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </i>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.5",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "badge": "",
    "classes": Array [
      "ha-icon",
      Object {
        "ha-icon-default": true,
      },
    ],
    "color": "",
    "dot": false,
    "imageIcon": false,
    "name": "default",
    "prefixCls": "ha-icon",
    "size": "default",
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
